### PR TITLE
Zoom in and zoom out feature for data visualizer (data editor canvas)

### PR DESCRIPTION
**Pull request by ThanatosGit**
I was so annoyed by the fact that you need to resize the window / `QDockWidget`s to scale, that I just wrote a quick zoom in and zoom out via mouse wheel.

### CODE REVIEW

**Review by henriquegemignani — APPROVED** [review]

**Comment by ThanatosGit** [comment]
Added another commit with a slider feature. It is a little bit tricky because of the different image sizes but I think this settings works for dread, super metroid and cave story.

File: randovania/gui/data_editor.py [hunk 1]
```diff
@@ -145,6 +147,20 @@ def __init__(self, data: dict, data_path: Path | None, is_internal: bool, edit_m
                                           RandovaniaGame.METROID_PRIME_CORRUPTION}:
             self.area_view_dock.hide()
 
+        # values for dread and super metroid
+        min_zoom = 7
```
Comment (henriquegemignani) [inline randovania/gui/data_editor.py:151]: I'd prefer that these values were to come from the game itself, not don't random constants here.
Maybe they're actually per-world?

File: randovania/gui/ui_files/ui_data_editor.py [hunk 2]
```diff
@@ -0,0 +1,261 @@
+# -*- coding: utf-8 -*-
```
Comment (henriquegemignani) [inline randovania/gui/ui_files/ui_data_editor.py:1]: This file is generated, shouldn't have been committed.

It's also been generated in the wrong path??

**Review by henriquegemignani — CHANGES_REQUESTED** [review]
Please also make sure to ensure all tests pass.

File: randovania/gui/data_editor.py (hunk 1 above)
Comment (ThanatosGit) [reply randovania/gui/data_editor.py:151]: Yeah, that's pretty dirty. The range for the scale variable really depends on the size / resolution of the images.
Where is a good place for these constants? The `extra` section in the JSONs? Like `Artaria.json` etc?
Than I could change it to be set by the `on_select_world` method.

File: randovania/gui/ui_files/ui_data_editor.py (hunk 2 above)
Comment (ThanatosGit) [reply randovania/gui/ui_files/ui_data_editor.py:1]: Yikes! Sorry! Maybe was due to misconfiguration at the beginning. I should have thought again why this generated file contained changes and was under version control

File: randovania/gui/data_editor.py (hunk 1 above)
Comment (henriquegemignani) [reply randovania/gui/data_editor.py:151]: Yeah, the extra section of each world is perfect.

Comment (henriquegemignani) [reply randovania/gui/data_editor.py:151]: There's already some logic to get the image per world or per area depending on some magical config. You could hook the same events to decide if these values would be per-world or per-area.

Comment (ThanatosGit) [reply randovania/gui/data_editor.py]: I now simply add the zoom from the slider to the already existing scale value. Not the best solution but still a little bit more comfort.

**Review by henriquegemignani — APPROVED** [review]

**Pull request merged**